### ISSUE TITLE
big number 15 significant digits error

### ISSUE THREAD

**Issue by okwme** [gnosis/dx-services#48]
```
yarn cli-rinkeby deposit 1.61744723307346186 WETH
yarn run v1.13.0
$ NETWORK=rinkeby node src/cli/cli.js deposit 1.61744723307346186 WETH
Keccak bindings are not compiled. Pure JS implementation will be used.
  INFO-dx-service:repositories:EthereumClient Checking Node connectivity... +0ms

  INFO-dx-service:repositories:EthereumClient Using Node Geth/v1.8.20-omnibus-c5febb98/linux-amd64/go1.11.1 +1s
  INFO-cli Deposit 1.617447233073462 WETH into the DX for 0x5899c1651653e1e4a110cd45c7f4e9f576de0670 +0ms
  INFO-dx-service:services:DxTradeService Fund the account 0x5899c1651653e1e4a110cd45c7f4e9f576de0670 with 1.617447233073462 WETH +0ms
cli.js deposit <amount> <token> [--account account]

Deposit funds in order to be used on the DX

Positionals:
  amount   Amount to buy, i.e. 0.8                                                        [required]
  token    Name of the token, i.e. WETH                                          [string] [required]
  account  Address where you send the tokens                                                [string]

Options:
  --version  Show version number                                                           [boolean]
  -h         Show help                                                                     [boolean]

{ BigNumber Error: lt() number type has more than 15 significant digits: 1617447233073462000
    at raise (/Users/billy/gnosis/dx-services/node_modules/bignumber.js/bignumber.js:1177:25)
    at new BigNumber (/Users/billy/gnosis/dx-services/node_modules/bignumber.js/bignumber.js:261:50)
    at BigNumber.P.lessThan.P.lt (/Users/billy/gnosis/dx-services/node_modules/bignumber.js/bignumber.js:1551:35)
    at DxTradeService._depositEtherIfRequired (/Users/billy/gnosis/dx-services/src/services/DxTradeService/DxTradeService.js:307:27)
    at processTicksAndRejections (internal/process/next_tick.js:81:5) name: 'BigNumber Error' }
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by johnerfx** [comment]
Hi,
I think this bug occurs because BigNumber(etherTokenBalance variable) is compared to normal number(amount variable)
https://github.com/gnosis/dx-services/blob/68e0ddae27ae567381c19cd1c0386f9f425ce789/src/services/DxTradeService/DxTradeService.js#L305-L307

Maybe line 307 should look like:
`if (etherTokenBalance.lessThan(amountInWei)) {`

But I am not sure, the same error occurs with other **cli commands** like `'buy'`, that is why you cannot use  your whole balance if it has more than 15digits. There can be more places in the code that behave the same, maybe the author can confirm this.

**Comment by dasanra** [comment]
Thank you for reporting.

BigNumber library is able to automatically convert Javascript Number into BigNumber to do operations, but this conversion has a 15 digit limitation. In order to avoid that is recommended to convert Javascript Number to String first.

In this PR we will change CLI commands to convert the amount passed to Wei using an internal util that checks this limitations and will pass an usable BigNumber even with an over 15 digit amount:

https://github.com/gnosis/dx-services/pull/49

This is planned to be fixed in `v2.0.3`

**Comment by johnerfx** [comment]
I think it might be necessary to bump yargs version too acording to comments below:
https://github.com/yargs/yargs/issues/550#issuecomment-460006293
> @TomFrost the specific issue you were running into should have been addressed by the fact that we now treat options that look like numbers as strings, if they're larger than JavaScript's `Number.MAX_SAFE_INTEGER`.

https://github.com/yargs/yargs/issues/1099

Otherwise more than 15 digits numbers could by rounded anyway while passing as CLI arguments.
By the way I am not sure about my observation here.
